src/internal.c

```diff
@@ -16114,22 +16114,30 @@ static int DoHandShakeMsgType(WOLFSSL* ssl, byte* input, word32* inOutIdx,
         case certificate_request:
         case server_hello_done:
             if (ssl->options.resuming) {
-#ifdef WOLFSSL_WPAS
+                /* https://www.rfc-editor.org/rfc/rfc5077.html#section-3.4
+                 *   Alternatively, the client MAY include an empty Session ID
+                 *   in the ClientHello.  In this case, the client ignores the
+                 *   Session ID sent in the ServerHello and determines if the
+                 *   server is resuming a session by the subsequent handshake
+                 *   messages.
+                 */
+                if (ssl->session->sessionIDSz != 0) {
+#ifndef WOLFSSL_WPAS
```

Comment:
when WOLFSSL_WPAS is defined this will be an empty if statement which may end up throwing warnings for some compilers

Reply:
Moved guard to encompass if statement.